### PR TITLE
Make upgrades more backup-friendly

### PR DESCRIPTION
**Pull request by Roy-Orbison**
Leave writable directories in place so backup systems know the files stored there have not changed.

### CODE REVIEW

**Comment by lindy65** [comment]
Thanks for the PR @Roy-Orbison!

Once you've signed the CLA we can get it through the review process 👍

**Comment by lindy65** [comment]
Hi @Roy-Orbison,

Wondering whether you've had a chance to sign our contribution licence agreement so we can go ahead and merge your PR?

**Comment by lindy65** [comment]
Thanks again for the PR but as we haven't heard from you regarding signing the CLA, I'll close this off for now.

**Comment by Roy-Orbison** [comment]
Signed it.

**Comment by lindy65** [comment]
Thanks @Roy-Orbison :)

We'll get your PR through the review process!  Thanks for working on this 👍

**Comment by Roy-Orbison** [comment]
I also added another commit to the same branch but it's not showing up here.

**Comment by jasonblais** [comment]
@Roy-Orbison Do you have a link to the commit - I'll see if I can help

**Comment by saturninoabril** [comment]
@Roy-Orbison I like the idea of making the upgrades more back-up friendly but I don't think this will work all the time.
Files and folder structures vary depending on version. Say for example: 
- version 4.1 has a folder `webapp`
- version 4.8 doesn't but there's `client` folder
- also the js files/maps  on those folders are different

If we are copying from upgrade to install folder, then the install folder will end up with unnecessary files. Old and new files/folders will remain on that install folder.

**Comment by Roy-Orbison** [comment]
@jasonblais It showed up, just took some time.

@saturninoabril Good point, I'll take another look.

**Comment by lindy65** [comment]
Hi @Roy-Orbison,

Just popping in to see if there's any update on this PR or anything you need help with?

**Comment by lindy65** [comment]
Hi @Roy-Orbison 

Just popping in to find out whether you're still interested in working on this PR?  Is there anything you need help with?

**Comment by amyblais** [comment]
Thank you for the PR but as we haven't heard from you for a while, I'll close this for now.

**Comment by Roy-Orbison** [comment]
Have had another crack, accounting for @saturninoabril's feedback.

**Comment by lindy65** [comment]
Thanks @Roy-Orbison,

We'll continue with the review process 👍

**Comment by saturninoabril** [comment]
Thanks @Roy-Orbison! I'll check this out first thing tomorrow.

File: source/administration/upgrade.rst [hunk 1]
```diff
@@ -56,25 +58,33 @@ Owner and group of the install directory - *{owner}* and *{group}*
...
 
-    ``sudo mv {install-path}/mattermost {install-path}/{mattermost-back-YYYY-MM-DD}``
+  .. code-block:: text
 
-7. Copy the files that you extracted earlier to the install directory.
+    cd {install-path}
+    sudo cp -ra mattermost/ mattermost-back-$(date +'%F-%H-%M')/
 
-  ``sudo cp -r mattermost {install-path}``
+7. Remove all files *except special directories* from within the current install directory.
```
Comment (saturninoabril) [inline source/administration/upgrade.rst]: I suggest to list those special directories.

**Comment by jasonblais** [comment]
Big thanks @Roy-Orbison! 

@amyblais We should get approvals from Derrick and Corey as well,

**Comment by Roy-Orbison** [comment]
Erm, this might need another update to include `plugins`. I only just noticed step 13's *Copy the `plugins/` directory from the old install to the new install*.

Do plugins have to be updated separately? If so, maybe the dir should be renamed (e.g. to `plugins~`) before, then back again after upgrade, so mattermost does not see the directory during, but its contents do not change.

**Pull request merged**